File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/SQLiteUpdateListener.java
package io.roastedroot.sqlite4j;

import org.jspecify.annotations.NonNull;

/** https://www.sqlite.org/c3ref/update_hook.html */
public interface SQLiteUpdateListener {

    public enum Type {
        INSERT,
        DELETE,
        UPDATE
    }

    void onUpdate(@NonNull Type type, @NonNull String database, @NonNull String table, long rowId);
}

File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/SQLiteUpdateListener.kt
package io.roastedroot.sqlite4j

/** https://www.sqlite.org/c3ref/update_hook.html  */
interface SQLiteUpdateListener {
    enum class Type {
        INSERT,
        DELETE,
        UPDATE
    }

    fun onUpdate(type: Type, database: String, table: String, rowId: Long)
}
